test/testdrive/materializations.td

```diff
@@ -139,16 +139,23 @@ a  b
 3  1
 1  2
 
-# Can create sink from unmaterialized view.
-> CREATE SINK not_mat_sink2 FROM data_view
+# Cannot create sink from unmaterialized view.
+! CREATE SINK not_mat_sink2 FROM data_view
   INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'data-view2-sink'
   FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
+contains:collection identifier is not present
 
 # Can create indexed view from unmaterialized view.
 > CREATE VIEW test5 AS
   SELECT b, max(a) AS c FROM data_view GROUP BY b
 > CREATE DEFAULT INDEX ON test5
 
+# or from an indexed unmaterialized view
+! CREATE SINK not_mat_sink2 FROM test5
+  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'data-view2-sink'
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
+contains:collection identifier is not present
```

Comment:
The error message is still not the must useful message. :sweat_smile:

Reply:
I'll make a TODO!

Reply:
https://github.com/MaterializeInc/materialize/issues/14221 !